packages/onyxsdk_pen/android/src/main/kotlin/com/example/onyxsdk_pen/OnyxsdkPenArea.kt

```diff
@@ -5,23 +5,110 @@ import android.graphics.Canvas
 import android.graphics.Color
 import android.graphics.Paint
 import android.graphics.Rect
-import android.util.Log
 import android.view.SurfaceView
 import android.view.View
 import com.onyx.android.sdk.data.note.TouchPoint
 import com.onyx.android.sdk.pen.RawInputCallback
 import com.onyx.android.sdk.pen.TouchHelper
 import com.onyx.android.sdk.pen.data.TouchPointList
+import com.onyx.android.sdk.api.device.epd.EpdController
+import com.onyx.android.sdk.api.device.epd.UpdateMode
 import io.flutter.plugin.platform.PlatformView
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.BinaryMessenger
+import io.flutter.plugin.common.MethodCall
 import java.util.Timer
 import java.util.TimerTask
+import androidx.annotation.NonNull
+
+internal class OnyxsdkPenArea(context: Context, messenger: BinaryMessenger, id: Int, creationParams: Map<String?, Any?>?) : PlatformView, MethodCallHandler {
+
+    enum class StrokeStyle(val Id: Int) {
+        FountainPen(0),
+        Pen(1),
+        Brush(2),
+        Pencil(3),
+        Marker(4)
+    }
+
+    private val channel: MethodChannel = MethodChannel(messenger, "onyxsdk_pen_area")
 
-internal class OnyxsdkPenArea(context: Context, id: Int, creationParams: Map<String?, Any?>?) : PlatformView {
     companion object {
         private val pointsToRedraw = 20
-        private val strokeWidth = 3f
     }
 
+    private var strokeWidth = 0.0f
+    private var strokeColor = Color.BLACK
+    private var strokeStyle = StrokeStyle.FountainPen
+
+    private fun updateStroke(paramsRef: Map<String, Any>?) {
+        /*
+         Flutter ints are variable width
+         Personally, I think it is utterly dumb. I hope there is a way to fix
+         int size in flutter (why would you need 64 bits to store srgb??),
+         but unless or until there isn't, this stupidity should be performed
+         */
+        val pureColor = (paramsRef?.get("strokeColor") as? Long)?.toInt()
+                              ?: paramsRef?.get("strokeColor") as? Int
+                              ?: Color.BLACK
```

Comment:
We have the [`fixnum`](https://pub.dev/packages/fixnum) package for fixed size ints, though I think I prefer this hack over polluting the dart code